File: dart/easy/countDuplicates.dart
class Item {
  late int item;
  late int count;
  late List<int> index;

  Item(this.item, this.count, this.index);
}

class CountDuplicates {
  static result(List<int> arr) {
    Map<String, dynamic> map = new Map<String, dynamic>();
    List result = [];

    for (int i = 0; i < arr.length; i++) {
      var elem = arr[i];
      if (map.containsKey('$elem')) {
        map['$elem']['count'] += 1;
        map['$elem']['index'].add(i);
      } else {
        map['$elem'] = {
          'item': elem,
          'count': 1,
          'index': [i],
        };
      }
    }

    map.forEach((key, value) {
      if (value['count'] > 1) {
        result.add(value);
      }
    });

    return result;
  }
}

File: js/easy/countDuplicates.js
function countDuplicates(arr) {
  var hashTable = {};
  var result = [];

  for (var i = 0; i < arr.length; i++) {
    if (hashTable[arr[i]]) {
      hashTable[arr[i]].count += 1;
      hashTable[arr[i]].index.push(i);
    } else {
      hashTable[arr[i]] = {
        item: arr[i],
        count: 1,
        index: [i],
      };
    }
  }

  for (var key in hashTable) {
    if (hashTable[key].count > 1) {
      result.push(hashTable[key]);
    }
  }
  return result;
}
//	{ item: 3, count: 2, index: [2,5], }
console.log(countDuplicates([1, 2, 3, 3, 4, 5, 5, 6, 7]));
console.log(countDuplicates([1, 2, 3, 4, 5, 6, 7]));

// O(n)
